softioc/builder.py

```diff
@@ -18,45 +18,52 @@
 # The SCAN field for all input records defaults to I/O Intr.
 
 
-def _in_record(record, name, **fields):
-    '''For input records we provide some automatic extra features: scanning,
-    initialisation as appropriate, and blocking puts from outside the IOC.'''
-
+def _set_in_defaults(fields):
     fields.setdefault('SCAN', 'I/O Intr')
-    if 'initial_value' in fields:
-        fields.setdefault('PINI', 'YES')
+    fields.setdefault('PINI', 'YES')
     fields.setdefault('DISP', 1)
-    return getattr(PythonDevice, record)(name, **fields)
+
+def _set_out_defaults(fields):
+    fields.setdefault('OMSL', 'supervisory')
+
+# For longout and ao we want DRV{L,H} to match {L,H}OPR by default
+def _set_scalar_out_defaults(fields, DRVL, DRVH):
```

Comment:
May just be my understanding, but "scalar" seems like the wrong word to use here? My assumption was this would apply to all non-Waveform Out records, as from my understanding they're all scalars.

Reply:
Yes, looking through the manual agrees with that. That's part of the reason I find the name confusing - to me "scalar" means any single-valued record, but this function takes it to mean analog or long records exclusively. Is this an EPICS convention I'm not aware of?

Reply:
I've looked up a couple of definitions of the word "scalar", and one of the defining properties of a "scalar" is that you can add values of this type (and multiply them, "scaling" them as it were).  It is not meaningful to add or scale booleans, enums or strings, so they're not "scalars"!

I just needed a simple shorthand name for this action; could have been `_set_drvl_drvh()` or similar.